src/controller/directoryController.ts

```diff
@@ -100,6 +101,25 @@ export class DirectoryController {
     }
   }
 
+  /**
+   * Deletes the associated uri of all selected AddonTreeItems.
+   * @param list Selected AddonTreeItems
+   * @returns the failed uris.
+   */
+  static async deleteUri(list: AddonTreeItem[]) {
+    const promises = [];
+    const failedUris: vscode.Uri[] = [];
+
+    for (const item of list) {
+      const thenable = vscode.workspace.fs.delete(item.uri, { recursive: true });
+        const promise = Promise.resolve(thenable).catch(() => failedUris.push(item.uri));
+        promises.push(promise);
```

Comment:
You can probably make that simpler: 

```suggestion
        const promise = vscode.workspace.fs.delete(item.uri, { recursive: true }).catch(() => failedUris.push(item.uri));
        promises.push(promise);
```

Reply:
Typescript gets mad since 'catch' doesn't exist on type 'Thenable<void>'. As far as I can tell Thenables _can_ do catches, but by using the `onRejected` parameter of .then() -- and this felt more legible in comparison.